taipy/core/scenario/scenario.py

```diff
@@ -175,22 +178,40 @@ def __hash__(self):
     def __eq__(self, other):
         return isinstance(other, Scenario) and self.id == other.id
 
-    def __getattr__(self, attribute_name):
-        protected_attribute_name = _validate_id(attribute_name)
-        if protected_attribute_name in self._properties:
-            return _tpl._replace_templates(self._properties[protected_attribute_name])
-
+    def __setattr__(self, name: str, value: Any) -> None:
+        if self.__CHECK_INIT_DONE_ATTR_NAME not in dir(self) or name in dir(self):
+            return super().__setattr__(name, value)
+        else:
+            protected_attribute_name = _validate_id(name)
+            try:
+                if protected_attribute_name not in self._properties and not self._get_attributes(
+                    protected_attribute_name, name
+                ):
+                    raise AttributeError
+                raise AttributeKeyAlreadyExisted(name)
+            except AttributeError:
+                return super().__setattr__(name, value)
+
+    def _get_attributes(self, protected_attribute_name, attribute_name) -> Union[Sequence, Task, DataNode]:
```

Comment:
The `protected_attribute_name` is not used outside of the `_get_attributes()` right?

We probably can move the `_validate_id()` inside the `_get_attributes()` to make it cleaner.
```suggestion
    def _get_attributes(self, attribute_name) -> Union[Sequence, Task, DataNode]:
        protected_attribute_name = _validate_id(name)
```

Reply:
well we need `_validate_id` in both __setattr__ and __getattr__ so it probably won't help much :/